presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -344,6 +344,12 @@ public synchronized int rewind(OptionalInt bucketNumber)
                         }
                         return revivedSplitCount;
                     }
+
+                    @Override
+                    public void decrementPartitionReferences(InternalHiveSplit split)
+                    {
+                        // no-op, we keep all splits forever, so references never decrease
```

Comment:
I don't have a strong opinion here, but this might make the semantic of this method ambiguous because we call `decrement` yet don't do anything. Maybe we could use `useRewindableSplitSource` to decide whether this method should be called, and throw unsupported exception here.